clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -92,6 +93,13 @@ public MetadataRequest build(short version) {
             if (!data.allowAutoTopicCreation() && version < 4)
                 throw new UnsupportedVersionException("MetadataRequest versions older than 4 don't support the " +
                         "allowAutoTopicCreation field");
+            if (data.topics() != null) {
+                data.topics().forEach(topic -> {
+                    if (topic.name() == null || topic.topicId() != Uuid.ZERO_UUID)
+                        throw new UnsupportedVersionException("MetadataRequest version " + version  +
```

Comment:
On second though, would it be better to have a separate exception message for each condition?

Reply:
sure